packages/dds/tree/src/simple-tree/core/treeNodeSchema.ts

```diff
@@ -316,6 +337,69 @@ export interface TreeNodeSchemaCore<
 	createFromInsertable(data: TInsertable): Unhydrated<TreeNode | TreeLeafValue>;
 }
 
+/**
+ * {@link TreeNodeSchemaCore} extended with some non-exported APIs.
+ */
+export interface TreeNodeSchemaCorePrivate<
+	Name extends string = string,
+	Kind extends NodeKind = NodeKind,
+	TInsertable = never,
+	ImplicitlyConstructable extends boolean = boolean,
+	Info = unknown,
+	TCustomMetadata = unknown,
+> extends TreeNodeSchemaCore<
+		Name,
+		Kind,
+		ImplicitlyConstructable,
+		Info,
+		TInsertable,
+		TCustomMetadata
+	> {
+	/**
+	 * All possible annotated allowed types that a direct child of a node with this schema could have, grouped by field.
+	 * If this node does not have fields, it will contain a single array with all its allowed types.
+	 *
+	 * Equivalently, this is also all schema directly referenced when defining this schema's allowed child types,
+	 * which is also the same as the set of schema referenced directly by the `Info` type parameter and the `info` property.
+	 * This property is simply re-exposing that information in an easier to traverse format consistent across all node kinds.
+	 * @remarks
+	 * Some kinds of nodes may have additional restrictions on children:
+	 * this set simply enumerates all directly referenced schema, and can be use to walk over all referenced schema types.
+	 *
+	 * This set cannot be used before the schema in it have been defined:
+	 * more specifically, when using lazy schema references (for example to make foreword references to schema which have not yet been defined),
+	 * users must wait until after the schema are defined to access this set.
+	 *
+	 * @privateRemarks
+	 * If this is stabilized, it will live alongside the childTypes property on {@link TreeNodeSchemaCore}.
+	 * @system
+	 */
+	readonly childAnnotatedAllowedTypes: readonly (readonly AnnotatedAllowedType<TreeNodeSchema>[])[];
```

Comment:
I think it would be better to use `AnnotatedAllowedTypes` here so we can actually access metadata at the AnnotatedAllowedTypes level (seems like when stuff eventually uses that, this should have access). It also makes the type a bit simpler.

Using AnnotatedAllowedTypes like this would mean that the type references could still be lazy (behind functions). I think that's ok since anyone who needs to can normalize them.

```suggestion
	readonly childAnnotatedAllowedTypes: readonly AnnotatedAllowedTypes)[];
```